java/core/src/main/java/com/microsoft/bond/StructBondTypeResolver.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.bond;
+
+/**
+ * Responsible for resolving generic Bond struct types.
+ * @param <TStruct> the class of the struct value
+ */
+public abstract class StructBondTypeResolver<TStruct extends BondSerializable> {
+
+    /**
+     * Resolves generic type arguments and creates a new instance of type descriptor that is not yet initialized.
+     * Generated sublasses add a public method to resolve and initialized the type descriptor.
+     *
+     * @param genericTypeArguments generic type arguments
+     * @return new unitialized type descriptor instance
+     */
+    protected abstract StructBondType<TStruct> resolveUninitialized(BondType<?>[] genericTypeArguments);
+
+    /**
+     * Resolves generic type arguments and creates and initializes new instance of type descriptor.
+     * Generated sublasses add a public method to resolve and initialized the type descriptor.
+     *
+     * @param genericTypeArguments generic type arguments
+     * @return new unitialized type descriptor instance
+     */
+    protected final StructBondType<TStruct> resolveAndInitialize(BondType<?>... genericTypeArguments) {
+        StructBondType<TStruct> type = this.resolveUninitialized(genericTypeArguments);
+        ensureInitialized(type);
```

Comment:
I am pretty sure it's fine as is.

---
In reply to: [123376279](https://github.com/Microsoft/bond/pull/502#discussion_r123376279) [](ancestors = 123376279)